logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/ephemeral/DeleteEphemeralMessageForSelfUserAsReceiverUseCase.kt

```diff
@@ -46,99 +46,100 @@ import com.wire.kalium.util.DateTimeUtil
  * for the self-deleting message, before the receiver does it on the sender side, the message is simply marked as deleted
  * see [com.wire.kalium.logic.feature.message.ephemeral.DeleteEphemeralMessageForSelfUserAsReceiverUseCaseImpl]
  **/
-interface DeleteEphemeralMessageForSelfUserAsReceiverUseCase {
+internal interface DeleteEphemeralMessageForSelfUserAsReceiverUseCase {
     /**
      * @param conversationId the conversation id that contains the self-deleting message
      * @param messageId the id of the self-deleting message
      */
     suspend operator fun invoke(conversationId: ConversationId, messageId: String): Either<CoreFailure, Unit>
 }
 
+@Suppress("LongParameterList")
 internal class DeleteEphemeralMessageForSelfUserAsReceiverUseCaseImpl(
     private val messageRepository: MessageRepository,
     private val assetRepository: AssetRepository,
+    private val ephemeralMessageRepository: EphemeralMessageRepository,
     private val currentClientIdProvider: CurrentClientIdProvider,
     private val messageSender: MessageSender,
     private val selfUserId: UserId,
     private val selfConversationIdProvider: SelfConversationIdProvider
 ) : DeleteEphemeralMessageForSelfUserAsReceiverUseCase {
     override suspend fun invoke(conversationId: ConversationId, messageId: String): Either<CoreFailure, Unit> =
-        messageRepository.getMessageById(conversationId, messageId).map { message ->
-            when (message.status) {
-                // TODO: there is a race condition here where a message can still be marked as Message.Status.FAILED but be sent
-                // better to send the delete message anyway and let it to other clients to ignore it if the message is not sent
-                Message.Status.FAILED -> messageRepository.deleteMessage(messageId, conversationId)
-                else -> {
-                    currentClientIdProvider().flatMap { currentClientId ->
-                        broadCastDeletionToConversation(messageId, conversationId, currentClientId)
-                            .flatMap {
-                                broadCastDeletionForSelfUser(messageId, conversationId, currentClientId)
-                            }
-                    }.onSuccess { deleteMessageAssetIfExists(message) }
-                        .flatMap { messageRepository.deleteMessage(messageId, conversationId) }
-                }
-            }.onFailure { failure ->
-                kaliumLogger.withFeatureId(MESSAGES).w("delete message failure: $message")
-                if (failure is CoreFailure.Unknown) {
-                    failure.rootCause?.printStackTrace()
+        currentClientIdProvider().flatMap { currentClientId ->
+            messageRepository.getMessageById(conversationId, messageId).flatMap { message ->
+                currentClientIdProvider().flatMap { currentClientId ->
```

Comment:
`currentClientIdProvider()` is used here twice, is it necessary or could we just leave only the first one?